src/explorer/CosmosDBTreeItem.ts

```diff
@@ -120,4 +117,89 @@ export class CosmosDBTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
     public hasMoreChildrenImpl(): boolean {
         return false;
     }
+
+    private detectMongoConnections(appSettings: { [propertyName: string]: string }): IDetectedConnection[] {
+        const result: IDetectedConnection[] = [];
+        for (const key of Object.keys(appSettings)) {
+            const value = appSettings[key];
+            if (/^mongodb[^:]*:\/\//i.test(value)) {
+                result.push({
+                    keys: [key],
+                    connectionString: appSettings[key]
+                });
+            }
+        }
+        return result;
+    }
+
+    private detectDocDBConnections(appSettings: { [propertyName: string]: string }): IDetectedConnection[] {
+        const result: IDetectedConnection[] = [];
+        const regexp = new RegExp(`(.+)${this._endpointSuffix}`, 'i');
+        for (const key of Object.keys(appSettings)) {
+            const match = key && key.match(regexp);
+            if (match) {
+                const prefix = match[1];
+                const endpointKey = prefix + this._endpointSuffix;
+                const keyKey = prefix + this._keySuffix;
+                const documentEndpoint: string | undefined = appSettings[endpointKey];
+                const masterKey: string | undefined = appSettings[keyKey];
+
+                if (documentEndpoint && masterKey) {
+                    const keys: string[] = [endpointKey, keyKey];
+                    let connectionString = `AccountEndpoint=${documentEndpoint};AccountKey=${masterKey};`;
+
+                    const databaseKey = prefix + this._databaseSuffix;
+                    if (Object.keys(appSettings).find((k) => k === databaseKey)) {
+                        keys.push(databaseKey);
+                        connectionString += `Database=${appSettings[databaseKey]}`;
+                    }
+
+                    result.push({ keys, connectionString });
```

Comment:
Looks good to me except I'm confused by this line.  I don't quite understand why you save all the keys as an array.  You only need to add the connection string, right?

Reply:
So the mongo scenario uses a single app setting for the connection string. However, the DocDB scenario uses three app settings:
```
AZURE_COSMOS_ENDPOINT
AZURE_COSMOS_MASTER_KEY
AZURE_COSMOS_DATABASE_ID
```

It does _not_ use the connection string directly. See Matt's sample project here: https://github.com/fiveisprime/node-todo/blob/cosmos/app/models/todo.js#L3